### PR TITLE
google place id 없이도 선택될 수 있도록

### PR DESCRIPTION
**Pull request by elzino**

### CODE REVIEW

**Comment by elzino** [comment]
selectAddress(address, placeId) {
-    this.clearAutocomplete()
-    this.handleSelect(address, placeId)
-    this.placesService.getDetails({placeId}, (place, status) => {
-      let changes = {address, google_place_id: placeId};
-      if (status === window.google.maps.places.PlacesServiceStatus.OK) {
-        const lat = place.geometry.location.lat();
-        const lng = place.geometry.location.lng();
-        const name = place.name;
-        const addr = place.formatted_address;
-        const timezone = place.utc_offset / 60;
-        changes = {...changes, lat, lng, name, addr, timezone};
-      }
+  selectAddress(address, placeId, item) {
+    this.clearAutocomplete();
+    this.handleSelect(address, placeId);
+    if (placeId) {
+      this.placesService.getDetails({placeId}, (place, status) => {
+        let changes = {address, google_place_id: placeId};
+        if (status === window.google.maps.places.PlacesServiceStatus.OK) {
+          const lat = place.geometry.location.lat();
+          const lng = place.geometry.location.lng();
+          const name = place.name;
+          const addr = place.formatted_address;
+          const timezone = place.utc_offset / 60;
+          changes = {...changes, lat, lng, name, addr, timezone};
+        }
 
-      this.handleDetailFetched(address, changes)
-    });
+        this.handleDetailFetched(address, changes);
+      });
+    } else {
+      const lat = +item.lat;
+      const lng = +item.lng;
+      const name = item.formattedSuggestion && item.formattedSuggestion.mainText;
+      const addr = item.formattedSuggestion && item.formattedSuggestion.secondaryText;
+      const timezone = item.timezone;
+      const info = {lat, lng, name, addr, address: addr, timezone};
+      this.handleDetailFetched(address, info);
+    }
   }

File: src/PlacesAutocomplete.js [hunk 1]
```diff
@@ -4,242 +4,251 @@
...
+    } else {
+      const lat = +item.lat;
+      const lng = +item.lng;
+      const name = item.formattedSuggestion && item.formattedSuggestion.mainText;
+      const addr = item.formattedSuggestion && item.formattedSuggestion.secondaryText;
+      const timezone = item.timezone;
+      const info = {lat, lng, name, addr, address: addr, timezone};
+      this.handleDetailFetched(address, info);
+    }
   }
```
Comment (elzino) [inline src/PlacesAutocomplete.js:141]: 여기요

File: src/PlacesAutocomplete.js [hunk 2]
```diff
@@ -249,123 +258,130 @@ class PlacesAutocomplete extends Component {
...
-          active: (highlightFirstSuggestion && idx === 0 ? true : false),
+          lat: location.latitude,
+          lng: location.longitude,
+          active: (!!(highlightFirstSuggestion && idx === 0)),
           index: idx,
           formattedSuggestion: formattedSuggestion(location),
-        }))
-      })
+          timezone: location.timezone,
+        })),
```
Comment (elzino) [inline src/PlacesAutocomplete.js:267]: 여기도 추가요

File: src/PlacesAutocomplete.js [hunk 3]
```diff
@@ -4,242 +4,251 @@
...
-        const lat = place.geometry.location.lat();
-        const lng = place.geometry.location.lng();
-        const name = place.name;
-        const addr = place.formatted_address;
-        const timezone = place.utc_offset / 60;
-        changes = {...changes, lat, lng, name, addr, timezone};
-      }
+  selectAddress(address, placeId, item) {
+    this.clearAutocomplete();
+    this.handleSelect(address, placeId);
```
Comment (gin1231) [inline src/PlacesAutocomplete.js:117]: 이거 아래 if 안으로 옮겨줘야할거같은데?

File: src/PlacesAutocomplete.js [hunk 4]
```diff
@@ -4,242 +4,251 @@
...
   handleEnterKey() {
-    const activeItem = this.getActiveItem()
+    const activeItem = this.getActiveItem();
     if (activeItem === undefined) {
-      this.handleEnterKeyWithoutActiveItem()
+      this.handleEnterKeyWithoutActiveItem();
     } else {
-      this.selectAddress(activeItem.suggestion, activeItem.placeId)
+      this.selectAddress(activeItem.suggestion, activeItem.placeId, activeItem);
     }
```
Comment (elzino) [inline src/PlacesAutocomplete.js:167]: 그리고 selectAddress에 추가로 item들을 다 추가 인자로 넘겻어요

File: src/PlacesAutocomplete.js (hunk 3 above)
Comment (elzino) [reply src/PlacesAutocomplete.js:117]: onAddressChange(inputText) {
    const {index} = this.props;
    this.props.updateLocationInfo(index, {
      uuid: this.props.item.uuid,
      name: inputText,
      locked: false,
    });
  }

Comment (gin1231) [reply src/PlacesAutocomplete.js:117]: ㅇㅋ 납두는거로

**Comment by gin1231** [comment]
👍

**Pull request merged**